server/src/functionalities/rename.ts

```diff
@@ -0,0 +1,68 @@
+import { RenameParams, TextDocuments, TextEdit, WorkspaceEdit } from 'vscode-languageserver'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import { Environment } from 'wollok-ts'
+import { Field, Node, Parameter, Reference, Variable } from 'wollok-ts/dist/model'
+import { cursorNode, fileNameToURI, toVSCRange } from '../utils/text-documents'
+import { referenceOf } from '../utils/vm/wollok'
+
+export const rename = (documents: TextDocuments<TextDocument>) => (environment: Environment) => (params: RenameParams): WorkspaceEdit | null => {
+  // cast cursor node as it's already validated in prepareRename request
+  const renamedNode = cursorNode(environment, params.position, params.textDocument) as Renamable | Reference<Renamable> | undefined
+  if(!renamedNode) throw new Error('No node found at position')
+
+  return {
+    changes: renamedNode.is(Reference) ?
+      groupByURI(renameNode(renamedNode.target!, params.newName, environment, documents)) :
+      groupByURI(renameNode(renamedNode, params.newName, environment, documents)),
```

Comment:
```suggestion
    changes: groupByURI(renameNode(renamedNode.is(Reference) ? renamedNode.target! : renamedNode, params.newName, environment, documents))
```

Y qué onda si no tiene target? (Una referencia rota)

Reply:
Saque todas las validaciones porque en `requestIsRenamable` ya se valida y **siempre** corre antes